openedx/core/djangoapps/course_groups/views.py

```diff
@@ -379,3 +390,356 @@ def debug_cohort_mgmt(request, course_key_string):
         kwargs={'course_key': text_type(course_key)}
     )}
     return render_to_response('/course_groups/debug.html', context)
+
+
+def _get_course_with_access(request, course_key_string, action='staff'):
+    """
+    Fetching a course with expected permission level
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    return course_key, get_course_with_access(request.user, action, course_key)
+
+
+def _get_cohort_response(cohort, course):
+    """
+    Helper method that returns APIView Response of a cohort representation
+    """
+    return Response(_get_cohort_representation(cohort, course), status=status.HTTP_200_OK)
+
+
+def _get_cohort_settings_response(course_key):
+    return Response(_cohort_settings(course_key))
+
+
+class APIPermissions(GenericAPIView):
+    authentication_classes = (JwtAuthentication, OAuth2AuthenticationAllowInactiveUser, SessionAuthentication,)
```

Comment:
Is there a reason why ``SessionAuthenticationAllowInactiveUser`` isn't included here?

Reply:
@davestgermain, that was an unintentional mistake and it has been addressed in the recent changes to this PR.